NineChronicles.Headless/GraphTypes/StateQuery.cs

```diff
@@ -40,18 +56,27 @@ public StateQuery()
                 resolve: context =>
                 {
                     var address = context.GetArgument<Address>("avatarAddress");
-                    try
-                    {
-                        return new AvatarStateType.AvatarStateContext(
-                            context.Source.AccountState.GetAvatarState(address),
-                            context.Source.AccountState,
-                            context.Source.BlockIndex);
-                    }
-                    catch (InvalidAddressException)
+                    return GetAvatarState(context.Source, address)
+                        ?? throw new InvalidOperationException($"The state {address} doesn't exists");
+                });
+            Field<NonNullGraphType<ListGraphType<AvatarStateType>>>(
+                name: "avatars",
+                description: "Avatar states having some order as addresses",
+                arguments: new QueryArguments(
+                    new QueryArgument<NonNullGraphType<ListGraphType<AddressType>>>
                     {
-                        throw new InvalidOperationException($"The state {address} doesn't exists");
+                        Name = "addresses",
+                        Description = "Addresses of agent to query avatars."
```

Comment:
It seems to be used as `avatar`'s addresses instead of `agent`'s addresses. Could you update the description?
```suggestion
                        Description = "Addresses of avatars to query."
```